include/pybind11/pybind11.h

```diff
@@ -883,19 +883,28 @@ class cpp_function : public function {
     }
 };
 
-/// Wrapper for Python extension modules
+/// Wrapper for Python modules
 class module_ : public object {
 public:
     PYBIND11_OBJECT_DEFAULT(module_, object, PyModule_Check)
 
     /// Create a new top-level Python module with the given name and docstring
-    PYBIND11_DEPRECATED("Use PYBIND11_MODULE or module_::create_extension_module instead")
     explicit module_(const char *name, const char *doc = nullptr) {
-#if PY_MAJOR_VERSION >= 3
-        *this = create_extension_module(name, doc, new PyModuleDef());
+#if defined(PYPY_VERSION)
```

Comment:
Sumitted https://foss.heptapod.net/pypy/pypy/-/merge_requests/769 to fix this

Reply:
Which got merged. So once we bump PyPy, this can be fixed.

Reply:
Should we wait with this PR until we bump PyPy?

Reply:
I don't think we should. This is really just a wrong signature (see PyPy PR) and PyPy isn't doing anything non-const, potentially dangerous with the `const_cast`ed string. We also don't know when PyPy will release 7.4.0, and whether we'll immediately drop support for 7.3.x, then. Meanwhile, the core of this PR is good.

I did add a more specific `(PYPY_VERSION_NUM < 0x07030400)` check, though. This should help when we do bump the minimum PyPy version supported.